core-bundle/src/Resources/contao/library/Contao/System.php

```diff
@@ -575,31 +575,37 @@ public static function isInstalledLanguage($strLanguage)
 	 * Return the countries as array
 	 *
 	 * @return array An array of country names
+	 *
+	 * @deprecated Deprecated since Contao 4.12, to be removed in Contao 5.
+	 *             Use the Symfony\Intl\Countries::getNames method instead.
 	 */
 	public static function getCountries()
 	{
-		$return = array();
-		$countries = array();
-		$arrAux = array();
+		trigger_deprecation('contao/core-bundle', '4.12', 'Using the %s method has been deprecated and will no longer work in Contao 5.0. Use the %s::getNames method instead', __METHOD__, Countries::class);
+
+		$request = self::getContainer()->get('request_stack')->getCurrentRequest();
+		$locale = $request ? $request->getLocale() : 'en';
+		$return = Countries::getNames($locale);
 
 		static::loadLanguageFile('countries');
-		include __DIR__ . '/../../config/countries.php';
 
-		foreach ($countries as $strKey=>$strName)
+		foreach ($return as $strKey=>$strName)
 		{
-			$arrAux[$strKey] = isset($GLOBALS['TL_LANG']['CNT'][$strKey]) ? Utf8::toAscii($GLOBALS['TL_LANG']['CNT'][$strKey]) : $strName;
+			if (isset($GLOBALS['TL_LANG']['CNT'][$strKey]))
+			{
+				$return[$strKey] = $GLOBALS['TL_LANG']['CNT'][$strKey];
+			}
 		}
 
-		asort($arrAux);
-
-		foreach (array_keys($arrAux) as $strKey)
-		{
-			$return[$strKey] = $GLOBALS['TL_LANG']['CNT'][$strKey] ?? $countries[$strKey];
-		}
+		(new \Collator($locale ?? 'en'))->asort($return);
 
 		// HOOK: add custom logic
 		if (isset($GLOBALS['TL_HOOKS']['getCountries']) && \is_array($GLOBALS['TL_HOOKS']['getCountries']))
 		{
+			trigger_deprecation('contao/core-bundle', '4.12', 'Using the "getCountries" hook has been deprecated and will no longer work in Contao 5.0.');
```

Comment:
Do we really want to deprecate this fully? I think it's worth keeping the possibility to change the list of countries in a central place.

Reply:
This should be adressed now with the `contao.intl.countries` config.